packages/react-reconciler/src/ReactFiberUnwindWork.js

```diff
@@ -246,6 +250,19 @@ function throwException(
               sourceFiber.tag = FunctionalComponent;
             }
 
+            if (
+              sourceFiber.tag === ClassComponent &&
+              workInProgress.stateNode === null
```

Comment:
Let's use `current === null` because we don't want to fire `componentWillUnmount` unless we fired `componentDidMount`.